src/rules/block-no-unnecessary-nesting/index.js

```diff
@@ -0,0 +1,77 @@
+"use strict";
+
+const resolveNestedSelector = require("postcss-resolve-nested-selector");
+const { utils } = require("stylelint");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("block-no-unnecessary-nesting");
```

Comment:
Hmm.. it looks like stylelint prefers to use `no-redundant` for the rules instead of `no-unnecessary`, so it would be better to follow the same convention.

Reply:
Renamed in [`d01ccc0`](https://github.com/stylelint-scss/stylelint-scss/pull/872/commits/d01ccc0afbae9d539ca634715bd5655ee40d33d1).